### ISSUE TITLE
No ownership and mode checks performed for binaries with capabilities

### ISSUE THREAD

**Issue by thoger** [rpminspect/rpminspect#1040]
**Is your feature request related to a problem? Please describe.**
This request is similar to what's noted in #1039 for setuid/setgid binaries, but for binaries with file capabilities.  For those, we should perform ownership and mode checks as well, not only capabilities list check.

However, unlike the problem for suid/sgid, this is also a design issue, not only implementation issue.  The capabilities allow-list format only contains the following information: package, path, and capabilities.  See: https://github.com/rpminspect/rpminspect/blob/v1.10/data/capabilities/GENERIC

The ownership and mode check is desired in cases like wireshark's /usr/bin/dumpcap, which is owned by root:wireshark and with mode 750 (i.e. only allowing wireshark group users to execute it).

**Describe the solution you'd like**
I'm open to discussion how to best address this - whether to require more information in capabilities allow-list, try to merge fileinfo and capabilities lists, or something else.

I think the current work around is to add any binary with capabilities to both fileinfo and capabilities lists.  The listing in fileinfo would ensure (once #1039 is fixed) ownership and mode checks.  This approach has drawbacks:

- The list of two files makes it likely to have incompletely or our-of-sync configuration.
- fileinfo listing is currently only required for suid/sgid, but not for capabilities.
- fileinfo is already meant to be used for other purposes - to allow-list not privileged binaries in bin_paths with different ownership than root:root.  Note that I'm not aware of any particular example of such binary and its listing in a fileinfo file.  For these not privileged binaries, there is no need to require security review.  But given that this case is not seen in the wild, the current status-quo seems acceptable.  Noting this explicitly so it can be kept in mind if designing any new allow-listing solution.

**Comment by dcantrell** [comment]
At this point I don't want to change the formats of the fileinfo or capabilities files.  They exist in the wild now and I don't want to have to go and change them and ensure backwards compatibility exists.

I think having ownership info recorded in fileinfo for files that also need entries in the capabilities file is the way to go.  You are right, I currently only use it for a few checks, but there's no reason it can't also be used for ownership checks in the capabilities inspection.

Thoughts?

**Comment by dcantrell** [comment]
So I'm looking at this one again and I don't see the concern.  The ownership inspection will check the ownership information, permissions will check permissions, and capabilities will check capabilities of a file.  I don't understand why the capabilities inspection needs to also check ownership and permissions -or- why that data needs to be in the same configuration file in the vendor data package.

The only thing I can see doing here is that if a file is listed in the capabilities file, it should also force the run of the ownership and permissions inspection for that entry.  Which will ensure you get all three checks even if the user has disabled permissions or ownership with -E or in the local config file.  Would that be sufficient?  rpminspect could also report if a fileinfo entry is missing for a file that is listed in the capabilities inspection as well.

**Comment by dcantrell** [comment]
Probably the least confusing things for users would be to further duplicate the permissions and ownership check within the capabilities inspection because automatically enabling those inspections if capabilities is enabled is likely to cause people to open bug reports.  What I was trying to do is avoid duplicating checks in the code, but I don't think it matters in this case.

For the example you mention where an executable gains or loses setuid or setgid during the lifetime of RHEL, that's the reason I have it read fileinfo files that match the product release string.  We can define as many of those as necessary in the vendor data package.  If /usr/bin/rm is setuid root starting in RHEL 9.6, then we can define a fileinfo file for el9.6 and add a new matching rule to the redhat.yaml file.  I believe I did that already but then the change was reverted or something.
